addons/controllers/antrea/antreaconfig_controller.go

```diff
@@ -0,0 +1,235 @@
+// Copyright YEAR VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	clusterapipatchutil "sigs.k8s.io/cluster-api/util/patch"
+
+	clusterapiutil "sigs.k8s.io/cluster-api/util"
+
+	"github.com/ghodss/yaml"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/constants"
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/util"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	addonsv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/addons/v1alpha1"
+)
+
+// AntreaConfigReconciler reconciles a AntreaConfig object
+type AntreaConfigReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=addons.tanzu.vmware.com,resources=antreaconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=addons.tanzu.vmware.com,resources=antreaconfigs/status,verbs=get;update;patch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// In Reconcile function, we compare the state specified by
+// the AntreaConfig object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+
+func (r *AntreaConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	log := r.Log.WithValues("antreaconfig", req.NamespacedName)
+
+	// get antrea config object
+	antreaConfig := &addonsv1alpha1.AntreaConfig{}
+	if err := r.Client.Get(ctx, req.NamespacedName, antreaConfig); err != nil {
+		log.Error(err, "unable to fetch AntreaConfig")
+		return ctrl.Result{}, err
```

Comment:
Do we want to requeue the request on NotFound errors? Shouldn't we return nil for such case? cc: @vijaykatam

Reply:
I think to answer the question, we need to understand when NotFound would happen. From my understanding, there is a very rare chance that would happen unless someone removes the finalizer and delete that obj very "quickly" between Lister gets the obj and controller reconciles. (Did I miss any other possibilities?)

Reply:
Fixed